Controller/Transaction/CreateWebpayM22.php

```diff
@@ -0,0 +1,138 @@
+<?php
+namespace Transbank\Webpay\Controller\Transaction;
+
+use Transbank\Webpay\Model\TransbankSdkWebpay;
+use Transbank\Webpay\Model\LogHandler;
+use \Transbank\Webpay\Model\Webpay;
+
+use \Magento\Sales\Model\Order;
+
+/**
+ * Controller for create transaction Webpay
+ */
+class CreateWebpayM22 extends \Magento\Framework\App\Action\Action {
+
+    public function __construct(\Magento\Framework\App\Action\Context $context,
+                                \Magento\Checkout\Model\Cart $cart,
+                                \Magento\Checkout\Model\Session $checkoutSession,
+                                \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
+                                \Magento\Quote\Model\QuoteManagement $quoteManagement,
+                                \Magento\Store\Model\StoreManagerInterface $storeManager,
+                                \Transbank\Webpay\Model\Config\ConfigProvider $configProvider) {
+
+        parent::__construct($context);
+
+        $this->cart = $cart;
+        $this->checkoutSession = $checkoutSession;
+        $this->resultJsonFactory = $resultJsonFactory;
+        $this->quoteManagement = $quoteManagement;
+        $this->storeManager = $storeManager;
+        $this->configProvider = $configProvider;
+        $this->log = new LogHandler();
+    }
+
+    public function execute() {
+
+        $response = null;
+        $order = null;
+        $config = $this->configProvider->getPluginConfig();
+        $orderStatusCanceled = $config['error_pay'];
+
+        try {
+
+            $guestEmail = isset($_GET['guestEmail']) ? $_GET['guestEmail'] : null;
+
+            $config = $this->configProvider->getPluginConfig();
+            $orderStatusPendingPayment = $config['order_status'];
+
+            $tmpOrder = $this->getOrder();
+
+            if ($tmpOrder != null && $tmpOrder->getStatus() == $orderStatusPendingPayment) {
+                $tmpOrder->setState($orderStatusCanceled)->setStatus($orderStatusCanceled);
+                $tmpOrder->save();
+                $this->checkoutSession->restoreQuote();
+            }
+
+            $quote = $this->cart->getQuote();
+
+            if ($guestEmail != null) {
+                $quote->getBillingAddress()->setEmail($guestEmail);
+                $quote->setData('customer_email', $quote->getBillingAddress()->getEmail());
```

Comment:
It's not possible to pass a hash to configure this, right?

Reply:
I don't know we can try this as an improvement in other PR.